pydoctor/model.py

```diff
@@ -183,22 +183,35 @@ def description(self) -> str:
         return self.module.fullName() if source_path is None else str(source_path)
 
     @property
-    def url(self) -> str:
-        """Relative URL at which the documentation for this Documentable
-        can be found.
+    def page_object(self) -> 'Documentable':
+        """The documentable to which the page we're documented on belongs.
+        For example methods are documented on the page of their class,
+        functions are documented in their module's page etc.
         """
         location = self.documentation_location
         if location is DocLocation.OWN_PAGE:
-            return f'{quote(self.fullName())}.html'
+            return self
         elif location is DocLocation.PARENT_PAGE:
             parent = self.parent
             if isinstance(parent, Module) and parent.name == '__init__':
                 parent = parent.parent
             assert parent is not None
-            return f'{quote(parent.fullName())}.html#{quote(self.name)}'
+            return parent
         else:
             assert False, location
 
+    @property
+    def url(self) -> str:
+        """Relative URL at which the documentation for this Documentable
+        can be found.
```

Comment:
Maybe use an explicit name ... so that you are not lost between absolute and relative URL all having URL.

And you can also have 2 types of "absolute URLs" : `/some/path/form/base.html`  vs 
`https://docs.example.com/some/path/form/base.html` 

```suggestion
    def relativeURL(self) -> str:
        """URL path and possible fragment for this Documentable,
        relative to the documentation base URL.
```

Reply:
I don't think we should be using absolute URLs at all: if we only use relative URLs, pydoctor's output can be plugged into a site at any depth, but if we use absolute URLs the user would have to specify the final location when they generate the docs.

Reply:
then maybe rename to `pathURL()` ?